var/spack/repos/builtin/packages/py-protobuf/package.py

```diff
@@ -71,7 +71,16 @@ class PyProtobuf(PythonPackage):
     depends_on("py-six@1.9:", when="@3:", type=("build", "run"))
     depends_on("py-ordereddict", when="@3: ^python@:2", type=("build", "run"))
     depends_on("py-unittest2", when="@3: ^python@:2", type=("build", "run"))
-    depends_on("protobuf", when="+cpp")
+
+    # Setup dependencies for protobuf to use the same minor version as py-protobuf
+    # Handle the 3.x series releases
+    for ver in list(range(1, 20)):
+        depends_on("protobuf@3." + str(ver) + ".0:3." + str(ver) + ".99",
+                   type=("build", "run"),
+                   when="+cpp @3." + str(ver) + ".0:3." + str(ver) + ".99")
```

Comment:
```suggestion
        depends_on("protobuf@3." + str(ver),
                   type=("build", "run"),
                   when="+cpp @3." + str(ver))
```

Reply:
@adamjstewart So there isn't always a one for one patch version that corresponds between the two packages.

Reply:
Yes, `@1.2` is equivalent to `@1.2.0:1.2.99` but technically more correct because it works even if there is a `1.2.100` release.

Reply:
@adamjstewart Okay thanks.